src/braket/emulators/emulator_interface.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+from abc import ABC
+from typing import Iterable, Union
+
+from braket.emulators.emulator_passes import EmulatorPass, ProgramType
+from braket.emulators.emulator_passes.criteria import EmulatorCriterion
+
+
+class EmulatorInterface(ABC):
+    def __init__(self, emulator_passes: Iterable[EmulatorPass] = None):
+        self._emulator_passes = emulator_passes if emulator_passes is not None else []
+
+    def run_program_passes(self, task_specification: ProgramType) -> ProgramType:
+        """
+        This method passes the input program through the EmulatorPasses contained
+        within this emulator. An emulator pass may simply validate a program or may
+        modify or entirely transform the program (to an equivalent quantum program).
+        Args:
+            task_specification (ProgramType): The program to run the emulator passes on.
+
+        Returns:
+            ProgramType: A "compiled" program of the same type as the input.
+
+        """
+        for emulator_pass in self._emulator_passes:
+            task_specification = emulator_pass(task_specification)
+        return task_specification
+
+    def run_validation_passes(self, task_specification: ProgramType) -> None:
+        """
+        This method passes the input program through EmulatorPasses that perform
+        only validation, without modifying the input program.
+
+        Args:
+            task_specification (ProgramType): The program to validate with this
+                emulator's validation passes.
+        """
+        for emulator_pass in self._emulator_passes:
+            if isinstance(emulator_pass, EmulatorCriterion):
```

Comment:
Why do we need this if-statement?

Reply:
In `validate()`, we only want to run the `EmulatorCriterion` emulator passes that are only used for validation, and are non-modifying; this check is used to avoid modifying `EmulatorPass` instances (currently, there are none, but if we have mapping/routing or nativization passes, we will want to avoid applying those to the Program when calling `validate()`).